pennylane/devices/default_qubit_jax.py

```diff
@@ -205,11 +205,18 @@ def _get_unitary_matrix(self, unitary):
             the unitary in the computational basis, or, in the case of a diagonal unitary,
             a 1D array representing the matrix diagonal.
         """
-        op_name = unitary.name
+        op_name = unitary.name.split(".inv")[0]
+
         if op_name in self.parametric_ops:
             if op_name == "MultiRZ":
-                return self.parametric_ops[unitary.name](*unitary.parameters, len(unitary.wires))
-            return self.parametric_ops[unitary.name](*unitary.parameters)
+                mat = self.parametric_ops[op_name](*unitary.parameters, len(unitary.wires))
+            else:
+                mat = self.parametric_ops[op_name](*unitary.parameters)
+
+            if unitary.inverse:
+                mat = self._transpose(self._conj(mat))
+
+            return mat

```

Comment:
I think we can both agree if we add any more transformations beyond `inv` that this method will get messy very quickly.

Reply:
Oh gosh yeah.

I think there are several things in the codebase we need to change:

* For gates where the inverse is 'known', treat `inv` as a tape transform, so that `RX(0.5).inv()` gets expanded to `RX(-0.5)`
* For gates where the inverse cannot be previously computed e.g., `S().inv()`, it gets passed to the device as-is

I think the latter case requires us to build support for `inv` directly into the device API. But I think `inv` is unique in this case - most transformations won't be low-level annotations that device frameworks implement natively.